### PR TITLE
Add draft attribute

### PR DESCRIPTION
**Pull request by gdraynz**
:warning: Rebased on top of #7

### CODE REVIEW

File: tukio/engine.py [hunk 1]
```diff
@@ -2,10 +2,10 @@
 Tukio Workflow Engine
 """
 import asyncio
-import weakref
 import logging
+import weakref
```
Comment (lhavel) [inline tukio/engine.py]: ?

File: tukio/engine.py [hunk 2]
```diff
@@ -87,6 +92,9 @@ def _load(self, template):
         Duplicates or invalid descriptions raise an exception.
         This operation does not affect workflow executions in progress.
         """
+        if template.draft:
+            raise DraftError('template must not be a draft')
```
Comment (lhavel) [inline tukio/engine.py]: I added a `LoadWorkflowError` in my PR TUKIO-4 which is more generic. I think we could use it here as well.

File: tukio/workflow.py [hunk 3]
```diff
@@ -137,11 +137,13 @@ class WorkflowTemplate(object):
...
     """
 
-    def __init__(self, title, uid=None, tags=None, version=None, policy=None):
+    def __init__(self, title, uid=None, tags=None, version=None,
+                 draft=None, policy=None):
         self.title = title
         self.tags = tags or []
         self.uid = uid or str(uuid4())
         self.version = int(version) if version is not None else 1
+        self.draft = bool(draft)
```
Comment (lhavel) [inline tukio/workflow.py]: Why `bool(draft)` and not draft. It should be `True` or `False` isn't it? With a default value to `False`.

File: tukio/engine.py (hunk 2 above)
Comment (lhavel) [inline tukio/engine.py]: As you prevent from loading draft workflows it won't be triggered by the engine.
How do you allow the user to play (=execute) with draft workflows.
Execute it outside of the engine? Or leave the user call `Engine.run` to trigger a draft workflow?

Comment (gdraynz) [inline tukio/engine.py]: I directly use `Workflow(template)` and `workflow.run()`, separated from the engine to keep the engine for production-ready workflows and use drafts as standalones.

File: tukio/engine.py (hunk 1 above)
Comment (gdraynz) [inline tukio/engine.py]: failed merge

File: tukio/workflow.py (hunk 3 above)
Comment (gdraynz) [inline tukio/workflow.py]: Used `bool()` to make sure it is always True or False.

**Comment by lhavel** [comment]
Diff embeds work from per-topic workflow selection and per-topic data receiving in tasks.
Let's recreate a clean PR.

**Pull request closed without merging**